lib/progress_bar.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Single-line CLI ASCII art progress bar
+class ProgressBar
+  attr_accessor :error, :steps
+  attr_reader :done
+
+  def initialize(owner = '', options = {})
+    @done = 0
+    @steps = 1
+    @owner = owner
+    @options = options
+    @error = false
+    @newline = false
+  end
+
+  def step!(obj, action = '')
```

Comment:
@moseshll not sure what I think about the variable name obj. is this more like the present step?

Reply:
It's just meant to be an integer. I think it's left over from my first looking at doing some kind of `def +()` operator. It's just meant to be the new integer that `@done` is set to. How would `  def step!(done, action = '')` look to you?